src/ceph-volume/setup.py

```diff
@@ -14,7 +14,10 @@
     keywords='ceph volume disk devices lvm',
     url="https://github.com/ceph/ceph",
     zip_safe = False,
-    install_requires='ceph',
+    install_requires=[
+        'ceph',
+        'backports.functools_lru_cache; python_version < "3.2.0"'
```

Comment:
I'm confused about whether we need `backports.functools_lru_cache` now, if we're not supporting py2 in master? Should we add that to the `nautilus` backport only?

Regardless, *wherever* we add `backports.functools_lru_cache`, let's make sure to capture that new dependency in the RPM spec and Debian packaging for Python 2.

Reply:
@ktdreyer the `backports.functools_lru_cache` is only needed for py2. I was adding it here conditionally, are you suggesting that `nautilus` backport be the only one with references to `backports.functools_lru_cache`? Does that make the packaging work easier if it's only mentioned in `nautilus`?

@jan--f I'm assuming you'll still want to cache `disk.get_devices` once #37274 is merged?

Reply:
For example [python-backports-functools_lru_cache](https://src.fedoraproject.org/rpms/python-backports-functools_lru_cache) is not available in the latest version of Fedora, since it's a Python 2-only package. If we don't support ceph-volume on py2 in Ceph master, and we are ok with making `nautilus` different than `master`, then we don't need `backports.functools_lru_cache` in master, and we can only add it in the backport.

The purpose of omitting it from master is that it keeps us from "carrying forward" this dependency.